### PR TITLE
zperf time handling fixes

### PR DESCRIPTION
**Pull request by jukkar**
Converted the code to use usec accuracy time functions that the kernel provides currently.
Also noticed IP address setting issue when testing the changes.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: samples/net/zperf/src/zperf_udp_uploader.c [hunk 1]
```diff
@@ -195,38 +209,44 @@ void zperf_udp_upload(const struct shell *shell,
...
+				 k_ticks_to_us_ceil64(loop_time -
+						      last_loop_time));
 		} else {
 			/* It's the first iteration so no need for adjustment
 			 */
 			adjust = 0;
 		}
 
-		if (adjust >= 0 || -adjust < delay) {
+		if (adjust >= 0 || -adjust < (int32_t)delay) {
```
Comment (pabigot) [inline samples/net/zperf/src/zperf_udp_uploader.c:232]: For minimal invocation of undefined behavior (sign/unsigned/conversion rank) I think this needs to be:

```
if (adjust >= 0) {
    delay += adjust
} else if ((uint64_t)-adjust < delay) {
    delay -= (uint64_t)-adjust;
} else {
    delay = 0U; /* ... */
}
```

**Review by pabigot — COMMENTED** [review]
Some comments related to upcoming restrictions of use of internal API, and signed/unsigned nit-picks.  I wish absolute times in ticks were represented as unsigned, but they're not.

**Comment by jukkar** [comment]
Updated according to comments.

**Comment by jukkar** [comment]
> I wish absolute times in ticks were represented as unsigned, but they're not.

I agree, the fact that some of the ticks/time/timeout functions use signed and some unsigned is quite confusing.

**Review by pabigot — APPROVED** [review]

**Pull request merged**